src/Aspire.Hosting/Extensions/ParameterResourceBuilderExtensions.cs

```diff
@@ -67,9 +77,34 @@ public static IResourceBuilder<IResourceWithConnectionString> AddConnectionStrin
         {
             return builder.Configuration.GetConnectionString(name) ?? throw new DistributedApplicationException($"Connection string parameter resource could not be used because connection string `{name}` is missing.");
         },
-        secret: true);
+        secret: true,
+        connectionString: true);
 
         var surrogate = new ResourceWithConnectionStringSurrogate(parameterBuilder.Resource, () => parameterBuilder.Resource.Value, environmentVariableName);
         return builder.CreateResourceBuilder(surrogate);
     }
+
+    /// <summary>
+    /// Changes the resource to be published as a connection string reference in the manifest.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <returns>The configured <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> PublishAsConnectionString<T>(this IResourceBuilder<T> builder)
+        where T : ContainerResource, IResourceWithConnectionString
```

Comment:
This is constrained to container resources so that we can avoid it showing up for child resources.